server/services/file_search_operations.py

```diff
@@ -1,14 +1,41 @@
 from io import BytesIO
+import re
+import os
 
-def extract_sections(file_path, search_terms, sections, specify_lines, use_total_lines,
-                     total_lines):
+def extract_sections(file_path, search_terms, sections, specify_lines, use_total_lines, total_lines):
     '''
     Extracts the data from orca log file based on search terms and sections.
     '''
     with open(file_path, 'r', encoding='utf-8') as f:
         lines = f.readlines()
 
     document_content = ""
+    
+    # Define the regex pattern for the header
+    header_pattern = r'^\s*NO\s+LB\s+ZA\s+FRAG\s+MASS\s+X\s+Y\s+Z\s*$'
+    
+    # Function to determine if a line is content
+    def is_content_line(line, term, header_pattern=None):
+        if line.strip() == "":
+            return False
+        if line.startswith("-----"):
+            return False
+        if line.startswith(term):
+            return False
+        if header_pattern and re.match(header_pattern, line):
+            return False
+        pattern = r'^\s*(\w+)\s+(\d+)\s+(\d+|\w+)\s+\w+'
```

Comment:
This regex is not working as expected. I added it for testing purposes. Can you please remove it as it is not part of this issue.